enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,258 @@
+settings.  If a test has its own unique test storage defined, then those
+settings would override the global storage settings.
+
+### Implementation Details/Notes/Constraints
+
+For each test that specifies storage, scorecard will interate
+this sequence for each test:
+ * scorecard adds a sidecar container into the test pod, the sidecar container would also mount the same emptyDir volume that is mounted into the test container
+ * upon test container completion and before removing the test pod, scorecard
+will extract (exec/tar) the test output contents from the sidecar container to the local host that scorecard is executing upon
+ * scorecard cleanup would cleanup the test Pod as normal, thereby also removing the added storage sidecar container
+
+A successful scorecard execution of tests that use this storage feature
+would include a local directory of the collected test output, with test output being divided into subdirectories based on suite and test names
+
+### Risks and Mitigations
+
+Scorecard users that don't require persistent storage are not impacted by this proposed feature.
+
+The storage sidecar would use the wait-time value to timeout in
+the event that scorecard abnormally exited before it performed 
+its test pod cleanup.
+
+## Design Details
+
+### emptyDir Volumes
+Each scorecard test that specifies the storage feature will have a unique emptyDir Volume into which they can write test output. This ensures that each
+test can be write output in isolation from other tests.  These volumes
+will be removed as part of scorecard's test Pod cleanup.  The presence of
+a storage spec within the scorecard config file triggers the addtion of the storage sidecar.
+
+By using emptyDir volumes for this feature, the cluster doesn't have to have a storage class defined and volume cleanup is included as part of test Pod cleanup.
+
+### Errors
+
+### Test Output
+Scorecard test output is gathered from the emptyDir volumes using a sidecar container that continues to run after the test container completes. The extraction of test output is performed by exec/tar into the sidecar container.
+Scorecard test output is extracted from the 'sidecar' container using client-go APIs (e.g. exec).  
+
+Test output is organized by test name in sub-directories.  For example, gathered output by default is stored in the local directory where
+scorecard was executed, however the user can also specify a command line flag (e.g. --test-output) to have the
+test output stored at a custom location.  The gathered output appears as follows:
+```bash
+./test-output
+./test-output/custom-suite/example-test1/somefile
+./test-output/custom-suite/example-test2/anotherfile
+./test-output/custom-suite/example-test3/foo.log
+```
+This directory will be over-written if scorecard is executed multiple times, it is therefore up to the end user
+to perform data management of this test output directory as they require.
```

Comment:
Is this going to be a problem? Would using a temporary directory with a unique number or timestamp be safer? i.e. `testoutput-20210416094443`? Maybe that's not super necessary. I know I'd be mad if I ran it twice and my original logs were wiped :)

Reply:
my thinking  was that the command flag --test-output could be used by the end user to store the output where they want it to be, somewhere they expect it to be for example.  If they wanted it to use a special naming convention to make it unique they could supply that using the CLI flag.